src/shared/parameter_v2/config_test.cpp

```diff
@@ -0,0 +1,257 @@
+#include <gtest/gtest.h>
+
+#include <boost/filesystem.hpp>
+#include <cstring>
+#include <regex>
+
+#include "clang-c/Index.h"
+#include "yaml-cpp/yaml.h"
+
+extern "C"
+{
+#include "shared/parameter_v2/c_parameters.h"
+}
+/**
+ * This function converts a config file names into config struct names.
+ *
+ * For example, a config file name maybe "example_config" gets converted
+ * to "ExampleConfig"
+ *
+ * @pre The snake_case_input must be valid snake_case
+ * @param The snake_case string to convert to CamelCase
+ * @return Converted string to CamelCase
+ */
+std::string toCamelCase(const std::string& snake_case_input)
+{
+    std::string ret_str = "";
+    ret_str += (char)toupper(snake_case_input[0]);
+
+    for (unsigned i = 1; i < snake_case_input.length(); i++)
+    {
+        if (snake_case_input[i] == '_')
```

Comment:
If the string ends with `_` this will fail trying to access the next character

Reply:
Fixed